lib/kino/table.ex

```diff
@@ -115,9 +119,14 @@ defmodule Kino.Table do
   end
 
   @impl true
-  def init({module, init_arg}, ctx) do
+  def init({module, init_arg, types}, ctx) do
     {:ok, info, state} = module.init(init_arg)
 
+    if types != nil and not Enum.all?(types, fn {_, value} -> value in @types end) do
+      raise ArgumentError,
+            "Invalid column types were provided: #{inspect(types)}. Valid types are: #{inspect(@types)}"
+    end
```

Comment:
Let's move this check to `new`, so it is validated before the process starts and should have a clearer stacktrace.

Reply:
Good catch! Fixed